src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdk.java

```diff
@@ -176,15 +181,49 @@ public SdkUpdater newUpdater() {
   }
 
   /** Get a new {@link ManagedCloudSdk} instance for @{link Version} specified. */
-  public static ManagedCloudSdk newManagedSdk(Version version) throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk(Version version)
+      throws UnsupportedOsException, InvalidOsStateException {
+    OsInfo osInfo = OsInfo.getSystemOsInfo();
     return new ManagedCloudSdk(
         version,
-        Paths.get(System.getProperty("user.home"), ".google-cloud-tools-java", "managed-cloud-sdk"),
-        OsInfo.getSystemOsInfo());
+        getOsSpecificManagedSdkHome(osInfo.name(), System.getProperties(), System.getenv()),
+        osInfo);
   }
 
   /** Convenience method to obtain a new LATEST {@link ManagedCloudSdk} instance. */
-  public static ManagedCloudSdk newManagedSdk() throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk()
+      throws UnsupportedOsException, InvalidOsStateException {
     return newManagedSdk(Version.LATEST);
   }
+
+  @VisibleForTesting
+  static Path getOsSpecificManagedSdkHome(
+      OsInfo.Name osName, Properties systemProperties, Map<String, String> environment)
+      throws InvalidOsStateException {
+    Path userHome = Paths.get(systemProperties.getProperty("user.home"));
+    Path cloudSdkPartialPath = Paths.get("google-cloud-tools-java", "managed-cloud-sdk");
+
+    switch (osName) {
+      case WINDOWS:
+        String localAppData = environment.get("LOCALAPPDATA");
+        if (localAppData == null || localAppData.trim().isEmpty()) {
+          throw new InvalidOsStateException("LOCALAPPDATA environment is invalid or missing");
+        }
+        return Paths.get(localAppData).resolve(cloudSdkPartialPath);
+
+      case MAC:
+        Path applicationSupport = userHome.resolve("Library").resolve("Application Support");
+        if (!Files.exists(applicationSupport)) {
+          throw new InvalidOsStateException(applicationSupport.toString() + " does not exist");
+        }
+        return applicationSupport.resolve(cloudSdkPartialPath);
+
+      case LINUX:
+        return userHome.resolve(".cache").resolve(cloudSdkPartialPath);
+
+      default:
+        // we can't actually get here
+        throw new RuntimeException();
```

Comment:
Couldn't this be `UnsupportedOsException`?

Reply:
This section is more programmer error, a consequence of me using a switch statement here?